src/proto/streams/prioritize.rs

```diff
@@ -744,6 +744,14 @@ impl Prioritize {
                             // capacity at this point.
                             debug_assert!(len <= self.flow.window_size());
 
+                            // Check if the stream level window the peer knows is available. In some
+                            // scenarios, maybe the window we know is available but the window which
+                            // peer knows is not.
+                            if len > 0 && !stream.send_flow.window_available(len) {
```

Comment:
I'd probably lean towards removing the `len > 0` part.

I see that `stream.send_flow.available()` is checked a few lines above, so the problem is that there's "available", but not enough `window_size`? Am I understanding the problem here?

(Wow, our choice of naming those fields could have been better. The current names don't distinguish what they really mean that well.)

Reply:
The "len > 0" part is for prevent zero-length frame from being blocked, which should not.